spring-data-jdbc/src/test/java/org/springframework/data/jdbc/repository/JdbcRepositoryIdGenerationIntegrationTests.java

```diff
@@ -29,31 +33,51 @@
 import org.springframework.context.annotation.FilterType;
 import org.springframework.context.annotation.Import;
 import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.PersistenceCreator;
+import org.springframework.data.annotation.Transient;
+import org.springframework.data.domain.Persistable;
 import org.springframework.data.jdbc.repository.config.EnableJdbcRepositories;
 import org.springframework.data.jdbc.repository.support.SimpleJdbcRepository;
+import org.springframework.data.jdbc.testing.AuditableNamedParameterJdbcTemplate;
 import org.springframework.data.jdbc.testing.IntegrationTest;
 import org.springframework.data.jdbc.testing.TestConfiguration;
 import org.springframework.data.relational.core.mapping.NamingStrategy;
+import org.springframework.data.relational.core.mapping.Sequence;
 import org.springframework.data.relational.core.mapping.event.BeforeConvertCallback;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.data.repository.ListCrudRepository;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.jdbc.Sql;
+import org.testcontainers.shaded.org.checkerframework.checker.index.qual.SameLen;
 
 /**
  * Testing special cases for id generation with {@link SimpleJdbcRepository}.
  *
  * @author Jens Schauder
  * @author Greg Turnquist
+ * @author Mikhail Polivakha
  */
 @IntegrationTest
 class JdbcRepositoryIdGenerationIntegrationTests {
 
-	@Autowired NamedParameterJdbcTemplate template;
-	@Autowired ReadOnlyIdEntityRepository readOnlyIdRepository;
+    @Autowired ReadOnlyIdEntityRepository readOnlyIdRepository;
 	@Autowired PrimitiveIdEntityRepository primitiveIdRepository;
 	@Autowired ImmutableWithManualIdEntityRepository immutableWithManualIdEntityRepository;
 
-	@Test // DATAJDBC-98
+    @Autowired EntityWithSeqRepository entityWithSeqRepository;
+
+    @Autowired PersistableEntityWithSeqRepository persistableEntityWithSeqRepository;
```

Comment:
There's quite some detail in the variable and repo name that doesn't contribute to the test fixture. I suggest something like `PersistableSeq. `Entity` is rather repetitive. Using inner classes for entities and repos allows to name them rather short (`PersistableSeqRepo`) and using shorter variable names ( `PersistableSeqRepo persistable`,`PrimitiveSeqRepo primitive`).

Reply:
I agree, good point. I'm actively learning from you, thank you for it :smile: